aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -75,34 +74,28 @@
     // Keeps track of whether of not the category is initialized yet
     private final CountDownLatch categoryInitializationsPending;
 
-    private final AtomicBoolean isOrchestratorReady;
-
-    // Used to interrogate plugins, to understand if sync should be automatically turned on
-    private final ApiCategory api;
-
     // User-provided configuration for the plugin.
     private final DataStoreConfiguration userProvidedConfiguration;
 
     // Configuration for the plugin that contains settings from the JSON file plus any
     // overrides provided via the userProvidedConfiguration
     private DataStoreConfiguration pluginConfiguration;
 
-    @SuppressLint("CheckResult")
     private AWSDataStorePlugin(
             @NonNull ModelProvider modelProvider,
             @NonNull ModelSchemaRegistry modelSchemaRegistry,
             @NonNull ApiCategory api,
             @Nullable DataStoreConfiguration userProvidedConfiguration) {
         this.sqliteStorageAdapter = SQLiteStorageAdapter.forModels(modelSchemaRegistry, modelProvider);
         this.categoryInitializationsPending = new CountDownLatch(1);
-        this.isOrchestratorReady = new AtomicBoolean(false);
-        this.api = api;
+        // Used to interrogate plugins, to understand if sync should be automatically turned on
         this.orchestrator = new Orchestrator(
             modelProvider,
             modelSchemaRegistry,
             sqliteStorageAdapter,
             AppSyncClient.via(api),
-            () -> pluginConfiguration
+            () -> pluginConfiguration,
+            api.getPlugins().isEmpty() ? Orchestrator.Mode.LOCAL_ONLY : Orchestrator.Mode.SYNC_VIA_API
```

Comment:
One issue I ran into here was that if I add the DataStore plugin before the API plugin, the Orchestrator goes into LOCAL_ONLY mode:
```
// Starts in LOCAL_ONLY
 Amplify.addPlugin(AWSDataStorePlugin())
Amplify.addPlugin(AWSApiPlugin())
```
as compared to:
```
// Starts in SYNC_VIA_API
Amplify.addPlugin(AWSApiPlugin())
Amplify.addPlugin(AWSDataStorePlugin())
```


 It's kind of a similar situation that we ran into for pluginConfiguration, where we ended up going with a provider-style configuration to delay evaluation of the expression.

Reply:
Ahhh, yup, absolutely. Great catch.